File: leap_year/ruby_leap_year/leap_year.rb
#!/usr/bin/ruby -w
#comments

def isLeapYear( year )
  if( year > 1581 ) then
    return(((year % 4 == 0) and (year % 100 > 0)) or (year % 400 == 0))
  else
    return false
  end
end

def main()
  if ARGV.length != 0 then
    puts "Usage: #{$0} <noargs>"
    exit(1)
  end
  #puts ARGV[0]
  for idx_i in (1999...2101)
    puts "isLeapYear(#{idx_i})=<#{isLeapYear(idx_i)}>"
  end
end

main()

File: leap_year/perl_leap_year/leap_year.pl
#!/usr/bin/perl -w

use strict;
use warnings;

sub main {
  my $idx_i;

  if( $#ARGV+1 != 0 ) {
    print STDERR "Usage: $0 <noargs>\n";
    exit(1);
  }

  for( $idx_i = 1999; $idx_i < 2101; $idx_i++ ) {
    print "isLeapYear(" . $idx_i . ")=<" . &isLeapYear($idx_i) . ">\n";
  }
  exit(0);
}

sub isLeapYear {
  my $year = shift;

  if( $year > 1581 ) {
    return((($year % 4 == 0) && ($year % 100 > 0)) || ($year % 400 == 0));
  } else {
    return 1;
  }
}

&main();
